diracx-core/src/diracx/core/exceptions.py

```diff
@@ -39,6 +39,15 @@ class InvalidQueryError(DiracError):
 
 
 class JobNotFound(Exception):
-    def __init__(self, job_id: int):
+    def __init__(self, job_id: int, detail: str | None = None):
         self.job_id: int = job_id
-        super().__init__(f"Job {job_id} not found")
+        super().__init__(f"Job {job_id} not found" + (" ({detail})" if detail else ""))
+
+
+class JobException(Exception):
```

Comment:
Based on https://github.com/DIRACGrid/diracx/pull/334, not merged yet:

```suggestion
class JobError(Exception):
```

Reply:
done!